palantirnet/federated-search-react
# FS-42 refactor global CSS

Completes: [FS-42](https://palantir.atlassian.net/browse/FS-42)

## To test:

Run `yarn start` to bring up app. Make sure it is configured to use autocomplete so that you can look at the autocomplete styles. You should compare the entire app at all breakpoints to the app before all CSS refactoring (look at commit 8a5ac306708b0d5181a9f698499b3b301f7d700d ) to make sure it visually matches. 

Note: I have already used [diffy](https://diffy.website/) to do this with a visual regression tool:

<img width="721" alt="Screen Shot 2020-02-26 at 2 51 44 PM" src="https://user-images.githubusercontent.com/6977052/75708525-d2747d00-5c75-11ea-8648-1f39b2489030.png">

Review the CSS for the following criteria:

- CSS styles are contained to the #fs-root element so that when this app is added as a module to other sites it will not alter the styling of those sites

- CSS is namespaced whenever possible using the fs- prefix.

- CSS components can be easily styled where you can use BEM and not need to include the main ID like `#fs-root div.myselector` 

- CSS in general seems maintainable and selectors can be easily overridden and a site's theme. Note that we have instructions that we will need to update [here](https://github.com/palantirnet/search_api_federated_solr/blob/108d46aa3596097858f19b8f21e3b03c91666e12/docs/theme.md) 


## Notes:
The main change here is that we had base element styles for font sizes, margins, padding combined with normalize CSS and I decoupled "base helpers" as placeholders from the actual "normalize" resets and browser/device fixes and then applied the placeholders as needed. One idea I'd like to chat about, is whether a normalize is even something we need, or can we assume most sites we are adding this to will have some kind of normalize. It would simplify things to remove it.
